### PR TITLE
Rom application numpy output process

### PR DESCRIPTION
**Pull request by SADPR**
**Description**
Created a process that outputs the Snapshots Matrix of the structural (displacements) and fluid dynamics (velocity and pressure) applications split into different ".npy" files. This helps for the first approach of ROM within eFlows4HPC project framework.

Please mark the PR with appropriate tags: 
- ROM 

**Changelog**
Summary:
E.g.
- tests/numpy_output_test_files: All files to run a test of this process
- custom_python/add_custom_utilities_to_python: Added missing lines for rom output utility
- python_scripts/numpy_output_process: Python NumPy output process outputs ".npy" files.
- custom_utilities/rom_output_utility: Loops over the nodes to obtain the snapshot of the current time/step. (This utility may be helpful for other incoming processes)

### CODE REVIEW

File: applications/RomApplication/custom_utilities/rom_output_utility.h [hunk 1]
```diff
@@ -0,0 +1,113 @@
+// KRATOS ┏━━━┓
```
Comment (rubenzorrilla) [inline applications/RomApplication/custom_utilities/rom_output_utility.h:1]: Implementation must be done in a cpp file.

File: applications/RomApplication/python_scripts/numpy_output_process.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import os
+
+def Factory(parameters, model):
+    if not isinstance(parameters, KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return NumpyOutputProcess(model, parameters["Parameters"])
+
+## All the processes python should be derived from "Process"
+
+class NumpyOutputProcess(KratosMultiphysics.Process):
```
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/numpy_output_process.py:13]: Shouldn't this derive from `OutputProcess`?

File: applications/RomApplication/tests/numpy_output_test_files/ProjectParameters.json [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        }],
+        "list_other_processes"     : [{
+            "python_module" : "numpy_output_process",
+            "kratos_module" : "KratosMultiphysics.RomApplication",
+            "Parameters"    : {
+                "main_model_part": "FluidModelPart",
+                "time_steps_per_file": 5,
+                "folder_name": "ObtainedOutput"
+            }
+        }],
```
Comment (rubenzorrilla) [inline applications/RomApplication/tests/numpy_output_test_files/ProjectParameters.json:102]: Shouldn't this block be in the `output_processes` list?

File: applications/RomApplication/custom_utilities/rom_output_utility.h (hunk 1 above)
Comment (SADPR) [reply applications/RomApplication/custom_utilities/rom_output_utility.h:1]: Sorry for the delay, I will do the corresponding modifications. Thanks.

**Pull request closed without merging**